plugins/view-resources/src/utils.ts

```diff
@@ -926,6 +939,86 @@ export async function getSpacePresenter (
   }
 }
 
+export async function getDocLabel (client: Client, object: Doc | undefined): Promise<string | undefined> {
+  if (object === undefined) {
+    return undefined
+  }
+
+  const hierarchy = client.getHierarchy()
+  const name = (object as any).name
+
+  if (name !== undefined) {
+    if (hierarchy.isDerived(object._class, contact.class.Person)) {
+      return getName(hierarchy, object as Contact)
+    }
+    return name
+  }
+
+  const label = hierarchy.getClass(object._class).label
+
+  if (label === undefined) {
+    return undefined
+  }
+
+  return await translate(label, {}, get(themeStore).language)
+}
+
+export async function getDocTitle (client: Client, object: Doc | undefined): Promise<string | undefined> {
+  if (object === undefined) {
+    return undefined
+  }
+
+  const hierarchy = client.getHierarchy()
+
+  const titleProvider = hierarchy.classHierarchyMixin(object._class, view.mixin.ObjectTitle)
+
+  if (titleProvider === undefined) {
+    return
+  }
+
+  const resource = await getResource(titleProvider.titleProvider)
```

Comment:
I suppose we need to pass _class and object into titleProvider to addition to object._id, so it will be possible to reused already fetched document.

Reply:
Done